akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -963,18 +963,23 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
                                  bufferPool: EnvelopeBufferPool): Sink[OutboundEnvelope, (OutboundCompressionAccess, Future[Done])] = {
 
     outboundLane(outboundContext, bufferPool)
-      .toMat(aeronSink(outboundContext, streamId))(Keep.both)
+      .toMat(aeronSink(outboundContext, streamId, bufferPool))(Keep.both)
   }
 
   def aeronSink(outboundContext: OutboundContext): Sink[EnvelopeBuffer, Future[Done]] =
     aeronSink(outboundContext, ordinaryStreamId)
 
-  private def aeronSink(outboundContext: OutboundContext, streamId: Int): Sink[EnvelopeBuffer, Future[Done]] = {
+  private def aeronSink(outboundContext: OutboundContext, streamId: Int): Sink[EnvelopeBuffer, Future[Done]] =
+    aeronSink(outboundContext, streamId, envelopeBufferPool)
```

Comment:
I think it's better to remove this overload and change at the call site (to make the parameter an explicit choice)